support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/builders/nameid/SamlProfileSamlNameIdBuilder.java

```diff
@@ -176,7 +181,10 @@ protected NameID finalizeNameId(final NameID nameid,
             if (StringUtils.isNotBlank(service.getNameIdQualifier())) {
                 nameid.setNameQualifier(service.getNameIdQualifier());
             } else {
-                val issuer = SamlIdPUtils.getIssuerFromSamlObject(authnRequest);
+                var issuer = SamlIdPUtils.getIssuerFromSamlObject(authnRequest);
+                if (StringUtils.equals(parseAndBuildRequiredNameIdFormat(service), NameIDType.PERSISTENT)) {
```

Comment:
This part cannot work. Idp entity id cannot be resolved statically from CAS configuration.

Reply:
And why is it working in my deployment? What do you mean by "statically"? Because it has to be configured in cas.properties to have a full working IdP? My guess was, that this is mandatory.
```
cas.authn.saml-idp.core.entity-id=https://idp.example.com/cas/idp
```

Reply:
It works for you, because likely you have a single idp instance. It will not work if the SP has an override for the idp, effectively turning CAS into a multi-tenant idp. And by statically, I mean exactly that.  Constantly. You cannot use a "static" value from the configuration and use that. It has to be resolved, dynamically, non-statically, at runtime.

Reply:
Sorry, but that's too much theory for me. I'm not a SAML pro either, but I've never seen a SP that overwrites the entityID of an IdP. I tested the whole thing via the test SPs of the DFN AAI (the German Federation), which let the user select the IdP via WAYF and it works without problems.
Could you give me a test scenario where the SP overwrites this?

Reply:
Sure. You'll find a few test cases for this case in `SamlProfileSamlAssertionBuilderTests`.

Reply:
It looks like it is impossible for me to solve the problem and thus find a general valid solution for better shibboleth SP compatibility.
I also understand that it is not possible for you in time to fix all the problems in CAS. But would it be possible to make a kind of work order between our institution and Apereo to solve these two problems (correct NameIDQualifier for persistent NameID Format & NameID Attribute Value)? We are happy to spend money on this, but need a contact person and a quote.
We lack developer experience here to find a solution in a reasonable time.

You can contact me via mail: robert.kornmesser@gfz-potsdam.de

Reply:
Sure. I'll send a you note shortly.